### ISSUE TITLE
features.to_csv() throws error 

### ISSUE THREAD

**Issue by Meijian** [jamesdolezal/slideflow#260]
## Description

I generated features with DatasetFeatures() function. When I tried to save slide-level features, it throws an error message. I tried tile-level and it didn't work either.

## To Reproduce
1.
  exm = build_feature_extractor('ctranspath', tile_px=299)
  features = sf.DatasetFeatures(exm, dataset=dataset)
2.
features.to_csv("/dbfs/mnt/gmb-ds-dbpoc/megu/aml88/ctranspath.activation.slide.csv", level='slide')

>>> features.to_csv("/dbfs/mnt/gmb-ds-dbpoc/megu/aml88/ctranspath.activation.tiles.csv", level='tile')
Working... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   0% -:--:--
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/local_disk0/.ephemeral_nfs/cluster_libraries/python/lib/python3.10/site-packages/slideflow/model/features.py", line 507, in to_csv
    if self.predictions[slide] != []:
KeyError: 'AVD-12TJG-1265-1'

## Environment:

 - Slideflow Version (e.g., 1.0): 2.0.3
 - OS (e.g., Ubuntu): Ubuntu
 - How you installed Slideflow (`pip`, source): pip
 - Python version: 3.9
 - CUDA/cuDNN version: 
 - GPU models and configuration:
 - Any other relevant information:

**Comment by jamesdolezal** [comment]
Thanks Meijian. While I investigate the to_csv() function in more detail, you should be able to workaround by first exporting to a pandas data frame, then exporting to CSV:

```
ftrs = DatasetFeatures(...)
ftrs.to_df().to_csv('/path/to/csv.')
```

**Comment by Meijian** [comment]
Thank you, James, I will try it. I can calculate myself but just wondering, does this workaround give slide-level features?

**Comment by jamesdolezal** [comment]
This gives tile-level features only - you would need to manually calculate slide-level features (just an average of tile-level features for each slide)

**Comment by Meijian** [comment]
Perfect, thank you James for your prompt actions!

**Comment by jamesdolezal** [comment]
This has been fixed in [2.0.4](https://github.com/jamesdolezal/slideflow/releases/tag/2.0.4).
